src/chains/ethereum/transaction/src/runtime-transaction.ts

```diff
@@ -24,7 +22,7 @@ import { TransactionReceipt } from "./transaction-receipt";
 import { Address } from "@ganache/ethereum-address";
 
 export const toValidLengthAddress = (address: string, fieldName: string) => {
-  const buffer = Data.from(address).toBuffer();
+  const buffer = Address.toBuffer(address);
   if (buffer.byteLength !== Address.ByteLength) {
```

Comment:
I don't think `Data.from(address).toBuffer()` should be changed to `Address.toBuffer(address)` in this case. I think this function is validating that the given string is already of the proper length (`Address.ByteLength`) and changing it to use `Address.toBuffer` will cause it to _always_ result in having a `byteLength` of `Address.ByteLength`.

The `toValidLengthAddress` function is used when validating the `from` and `to` address of a JSON transaction object. I believe we need to ensure that the `from` and `to` addresses are always 40 hex characters.

Reply:
Good call - I've changed this to `Data.toBuffer()` instead.

Reply:
FYI: latest code still shows this line as `Addres.toBuffer(address)`

Reply:
Yeah, I ran into an issue last night where this was causing test failures, so left it for today. That's why I converted to draft!